test/int32.test.js

```diff
@@ -0,0 +1,261 @@
+'use strict';
+
+const assert = require('assert');
+const start = require('./common');
+
+const mongoose = start.mongoose;
+const Schema = mongoose.Schema;
+
+describe('Int32', function() {
+  beforeEach(() => mongoose.deleteModel(/Test/));
+
+  it('is a valid schema type', function() {
+    const schema = new Schema({
+      myInt32: Schema.Types.Int32
+    });
+    const Test = mongoose.model('Test', schema);
+
+    const doc = new Test({
+      myInt32: 13
+    });
+    assert.strictEqual(doc.myInt32, 13);
+    assert.equal(typeof doc.myInt32, 'number');
+  });
+
+  it('supports the \'required\' property', async function() {
+    const schema = new Schema({
+      int32: {
+        type: Schema.Types.Int32,
+        required: true
+      }
+    });
+    const Test = mongoose.model('Test', schema);
+
+    const doc = new Test({
```

Comment:
It might be worth adding a test for a non-null required value as well, simply because local testing of your PR caught that:

```typescript
new Test({ 
	int: 3
})
```

still throws, because your `checkRequired` function is expecting an instance of our BSON Int32 class